Modules/src/UserInputValidator.cc

```diff
@@ -44,6 +61,86 @@ namespace ChimeraTK {
     return rejected;
   }
 
+  void UserInputValidator::enableDeepValidation() {
+    // Find out accessors with return
+    for(auto& accessor : _module->getAccessorListRecursive()) {
+      if(accessor.getDirection() == VariableDirection{VariableDirection::feeding, true} &&
+          accessor.getModel().getTags().count(std::string(UserInputValidator::tagValidatedVariable)) > 0) {
+        _downstreamInvalidatingReturnChannels.emplace(accessor.getAppAccessorNoType().getId());
+      }
+    }
+
+    auto keepPvAccesWithReturnChannel = Model::EdgeFilter([](const Model::EdgeProperties& edge) -> bool {
+      return edge.type == Model::EdgeProperties::Type::pvAccess && edge.pvAccessWithReturnChannel;
+    });
+
+    // Filter that keeps our tagged variable nodes and application modules
+    auto validatedVariablesAndApplicationModulesFilter =
```

Comment:
Why don't you use `keepTag(UserInputValidator::tagValidatedVariable) || keepApplicationModules()` instead?

Reply:
Can be removed completely